pkg/order/mempool/mempool_impl.go

```diff
@@ -91,10 +101,17 @@ func (mpi *mempoolImpl) processDirtyAccount(dirtyAccounts map[string]bool) {
 			readyTxs, nonReadyTxs, nextDemandNonce := list.filterReady(pendingNonce)
 			mpi.txStore.nonceCache.setPendingNonce(account, nextDemandNonce)
 
-			// inset ready txs into priorityIndex.
+			// insert ready txs into priorityIndex and set ttlIndex for these txs.
 			for _, tx := range readyTxs {
-				mpi.txStore.priorityIndex.insertByOrderedQueueKey(account, tx)
+				if !mpi.txStore.priorityIndex.data.Has(makeTimeoutKey(account, tx)) {
+					mpi.txStore.priorityIndex.insertByTimeoutKey(account, tx)
+					if list.items[tx.Nonce].local {
+						// no need to rebroadcast tx from other nodes to reduce network overhead
+						mpi.txStore.ttlIndex.insertByTtlKey(account, tx.Nonce, tx.Timestamp)
+					}
```

Comment:
the insertByTtlKey() func shoud be called in method insertTxs, ttlIndex is not directly related to the ready queue.